src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -647,42 +648,182 @@ public void Fault(RequestChannel channel)
+                                }
+                                else
+                                {
+                                    _httpRequestMessage.Headers.TransferEncoding.TryParseAdd(value);
+                                }
+                            }
+                            else if (string.Compare(name, "user-agent", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                _httpRequestMessage.Headers.UserAgent.Add(ProductInfoHeaderValue.Parse(value));
+                            }
+                            else if (string.Compare(name, "if-modified-since", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                DateTimeOffset modifiedSinceDate;
+                                if (DateTimeOffset.TryParse(value, DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AllowWhiteSpaces | DateTimeStyles.AssumeLocal, out modifiedSinceDate))
+                                {
+                                    _httpRequestMessage.Headers.IfModifiedSince = modifiedSinceDate;
+                                }
+                                else
+                                {
+                                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
+                                        new ProtocolException(SR.Format(SR.HttpIfModifiedSinceParseError, value)));
+                                }
+                            }
+                            else if (string.Compare(name, "date", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                // this will be taken care of by System.Net when we make the request
+                            }
+                            else if (string.Compare(name, "proxy-connection", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                throw ExceptionHelper.PlatformNotSupported("proxy-connection");
+                            }
+                            else if (string.Compare(name, "range", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                // we don't support ranges in v1.
+                            }
+                            else
+                            {
+                                if (!_httpRequestMessage.Headers.TryAddWithoutValidation(name, value))
+                                {
+                                    throw FxTrace.Exception.AsError(new InvalidOperationException(SR.Format(
+                                                    SR.CopyHttpHeaderFailed,
+                                                    name,
+                                                    value,
+                                                    _httpRequestMessage.Headers.GetType().Name)));
+                                }
```

Comment:
@roncain, I like your idea of not using the Try variant. Right now there's no way to know what the problem was. It's possible the Add will give a more meaningful exception. We still need to wrap it in an InvalideOperationException though for consistent exception handling, and the resource string still requires a type name be provided.